swap/swap.go

```diff
@@ -491,8 +532,8 @@ func (s *Swap) deployLoop(opts *bind.TransactOpts, owner common.Address, default
 			time.Sleep(deployDelay)
 		}
 
-		if s.contract, tx, err = contract.Deploy(opts, s.backend, owner, defaultHarddepositTimeoutDuration); err != nil {
-			auditLog.Warn("can't send chequebook deploy tx", "try", try, "error", err)
+		if addr, tx, _, err = contract.Deploy(opts, s.backend, owner, defaultHarddepositTimeoutDuration); err != nil {
```

Comment:
yes, this is production code, not test code, let's not ignore, and if it needs to be ignored, explain why

Reply:
I wouldn't worry too much about it now. This will be changed with the factory PR anyway.

Reply:
`contract.deploy` returns an address, the transaction identifier, and a simpleSwap instance. 

The simpleSwap instance is redundant here because the `startChequebook` function ends with a call to `bindToContractAt`, which sets the contract instance on `swap`.

I think this design is cleaner because the code now clearly does what it says it does: `deploy` deploys and only deploys, `bindToContractAt` binds the contract to SWAP and nothing else.

Furthermore, `startChequebook` is easy to reason about now: the function body gets an address of a chequebook (either by deploying or pass-in), it ends with binding (which both verifies the deployed code and saves the chequebook for later usage).

Reply:
I will remove the `simpleSwap` instance as a return variable from `contract.deploy`

Reply:
Changed it not as described above. We are now returning the *instance* from the whole deploy structure, and use this instance as a return variable from `startChequebook`.